src/sentry/deletions/defaults/grouphash.py

```diff
@@ -1,11 +1,24 @@
 from sentry.deletions.base import BaseRelation, ModelDeletionTask, ModelRelation
 from sentry.models.grouphash import GroupHash
+from sentry.models.grouphashmetadata import GroupHashMetadata
 
 
 class GroupHashDeletionTask(ModelDeletionTask[GroupHash]):
     def get_child_relations(self, instance: GroupHash) -> list[BaseRelation]:
-        from sentry.models.grouphashmetadata import GroupHashMetadata
-
         return [
             ModelRelation(GroupHashMetadata, {"grouphash_id": instance.id}),
         ]
+
+    def delete_instance(self, instance: GroupHash) -> None:
+        # Prevent integriy errors in cases where a hash which seer matched to this one has since
+        # been unmerged from the group, and the new group isn't getting deleted alongside
+        # `instance`'s group.
+        #
+        # (In cases where no unmerging has happened, relevant `GroupHashMetadata` objects will
+        # already have been deleted by the time we get here, since their grouphashes point to the
+        # same group as `instance` does.)
+        GroupHashMetadata.objects.filter(seer_matched_grouphash=instance).update(
```

Comment:
do you know if there's an index on `seer_matched_grouphash`? (if its a foreign key one is created automatically). just want to make sure this query runs quickly

Reply:
> if its a foreign key one is created automatically

It is indeed a foreign key, so if the above is true, there should be one. How would I check that to confirm?